File: app/src/main/java/com/example/tribecrypto/repository/CurrencyDetailsRepository.java
package com.example.tribecrypto.repository;

import com.example.tribecrypto.data.entity.CryptoCurrencyDetailsEntity;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;

public interface CurrencyDetailsRepository {

    Observable<List<CryptoCurrencyDetailsEntity>> getAllCurrenciesDetails();
    Completable insertDetailsList(List<CryptoCurrencyDetailsEntity> data);
    Completable updateDetailsList(List<CryptoCurrencyDetailsEntity> data);
    Completable removeAllDetails();
}

File: app/src/main/java/com/example/tribecrypto/repository/CurrencyDetailsRepository.kt
package com.example.tribecrypto.repository

import com.example.tribecrypto.data.entity.CryptoCurrencyDetailsEntity
import io.reactivex.Completable
import io.reactivex.Observable

interface CurrencyDetailsRepository {

    fun getAllCurrenciesDetails(): Observable<List<CryptoCurrencyDetailsEntity>>
    fun insertDetailsList(data: List<CryptoCurrencyDetailsEntity>): Completable
    fun updateDetailsList(data: List<CryptoCurrencyDetailsEntity>): Completable
    fun removeAllDetails(): Completable
}